internal/builder/packages.go

```diff
@@ -341,17 +368,23 @@ func createBuildDirectory(dirs ...string) (string, error) {
 	return buildDir, nil
 }
 
-func findRepositoryLicense(licenseTextFileName string) (string, error) {
-	dir, err := files.FindRepositoryRootDirectory()
-	if err != nil {
-		return "", err
-	}
-
-	sourceFileName := filepath.Join(dir, licenseTextFileName)
-	_, err = os.Stat(sourceFileName)
-	if err != nil {
+// findRepositoryLicensePath checks if a license file exists at the specified path.
+// If the file exists, it returns the path; otherwise, it returns an error indicating
+// that the repository license could not be found.
+//
+// Parameters:
+//
+//	licensePath - the path to the license file.
+//
+// Returns:
+//
+//	string - the license file path if found.
+//	error  - an error if the license file does not exist.
+func findRepositoryLicensePath(repoRoot *os.Root, repositoryLicenseTextFileName string) (string, error) {
+	path := filepath.Join(repoRoot.Name(), repositoryLicenseTextFileName)
+	if _, err := os.Stat(path); err != nil {
```

Comment:
Could it be used here `repoRoot.Stat()`? 

```suggestion
	if _, err := repoRoot.Stat(repositoryLicenseTextFileName); err != nil {
```

Reply:
in this case, we could use repoRoot.Stat as `repositoryLicenseTextFileName ` is a filename and not an absolute path.